tests/functional/myst/doctest.md

```diff
@@ -0,0 +1,16 @@
+# DocTest examples
+
+A Python REPL / DocTest example using normal way of specifying Python
+language:
+
+```python
+>>> 1+1
+2
+```
```

Comment:
@hynek - is this how you do doctest-style Python REPL stuff in MyST or is there a better way?

Reply:
sorry for the delay…the current way to do doctest in Sphinx + MyST looks like this:

    ```{eval-rst}
    .. doctest::

       >>> from structlog import get_logger
       >>> from structlog.testing import capture_logs
       >>> with capture_logs() as cap_logs:
       ...    get_logger().bind(x="y").info("hello")
       >>> cap_logs
       [{'x': 'y', 'event': 'hello', 'log_level': 'info'}]
    ```

So if you implement that, you'd be drop-in.

---

I hope you might be able to work with the MyST devs to give us

    ```{doctest}

or

    ```{sybil}

for that matter?

Reply:
If you try:

    ```python
       >>> from structlog import get_logger
       >>> from structlog.testing import capture_logs
       >>> with capture_logs() as cap_logs:
       ...    get_logger().bind(x="y").info("hello")
       >>> cap_logs
       [{'x': 'y', 'event': 'hello', 'log_level': 'info'}]
    ```

I think you'll find the pygments rendering does the right thing, so I'll have sybil support that.

If you use:

    ```{eval-rst}
    .. doctest::

       >>> from structlog import get_logger
       >>> from structlog.testing import capture_logs
       >>> with capture_logs() as cap_logs:
       ...    get_logger().bind(x="y").info("hello")
       >>> cap_logs
       [{'x': 'y', 'event': 'hello', 'log_level': 'info'}]
    ```

...then you won't have to change anything, but just use Sybil's resting `sybil.parsers.rest.DocTestParser` to pick them out.

I'm afraid I don't have bandwidth to contribute to MyST, but if they add a `{doctest}` role, then adding support to Sybil will be trivial.

Reply:
By "work" I meant "ask them for it". :) I think they'd be happy to get rid of every instance of `{eval-rst}`

Reply:
Why wouldn't you use the python-code-block-with-repl-in-it which works already?

        A Python REPL / DocTest example using normal way of specifying Python
        language:
        
        ```python
        >>> 1+1
        2
        ```

...renders as:
<img width="697" alt="Screenshot 2022-11-01 at 13 25 42" src="https://user-images.githubusercontent.com/632049/199231899-7528416d-63b8-4626-a3ad-727f1e235862.png">

The "doctest" role in ReST looks like a thing invented by `sphinx.ext.doctest`, which isn't needed for Sybil (although stuff written in that style will "just work"), so I'd expect the MyST folks to say it isn't needed for them either.

Reply:
That's true but I have tons of code blocks like that that aren't valid/runnable doctests. Assuming every Python code block that looks like a session will break A LOT of builds.

Reply:
Can you point me at some examples of those? It'd be interesting to think about the best way to handle them...

Reply:
On a phone rn but imagine someone wants to demonstrate how easy httpx is:

    ```Python
    >>> httpx.get("whatever")
    ...
    ```

The first problem is that you're making network calls. :)

Reply:
Sure, but that's the same as anything like that which you want to test: you need to mock... here's how I did it for [configurator](https://configurator.readthedocs.io/en/latest/):

https://github.com/simplistix/configurator/blob/master/conftest.py
https://github.com/simplistix/configurator/blame/master/README.rst#L28-L36

For existing stuff, you can add [skips](https://sybil.readthedocs.io/en/latest/parsers.html#skip) (which I'll definitely be adding in the MyST support!).

That said, if you have some real-world existing examples, it'd be interesting to see if a custom parser or some parser option could be added to ease migration.

Reply:
I'm bearing great news! According to @Tinche

    ```{doctest}

already works!

c.f. https://github.com/python-attrs/cattrs/blob/tin/tagged-unions/docs/strategies.md

So how about being conservative and support only those two use-cases for now? 😇